### PR TITLE
[radio] add tx timestamp

### PR DESCRIPTION
**Pull request by bukepo**
This commit adds tx timestamp so that the pcap callback can get the time when a frame was transmitted. This commit also adds channel, LQI and timestamp info in pcap handler in ncp.

### CODE REVIEW

**Comment by bukepo** [comment]
If putting the `mTimestamp` ahead, there will be significant code size increase in ARM architecture, which is caused by the instruction change from `STRB` -> `STRB.w` in many functions to access last fields(fields with offset larger than 32 bytes, thanks to @erjiaqing for help on analysis) in `otRadioFrame`. Putting the less frequently used field `mTimestamp` in the end avoids the problem.

Besides, there are 16 receive buffers in *nRF52840* platform, so the structure size increase is amplified 16 times, which result in the outstanding increase in `bss` section.

File: include/openthread/platform/radio.h [hunk 1]
```diff
@@ -361,6 +361,12 @@ typedef struct otRadioFrame
             bool mCsmaCaEnabled : 1;       ///< Set to true to enable CSMA-CA for this packet, false otherwise.
             bool mCslPresent : 1;          ///< Set to true if CSL header IE is present.
             bool mIsSecurityProcessed : 1; ///< True if SubMac should skip the AES processing of this frame.
+
+            /**
+             * The time of the local radio clock in microseconds when the end of
+             * the SFD was present at the local antenna.
+             */
+            uint64_t mTimestamp;
```
Comment (abtink) [inline include/openthread/platform/radio.h:369]: Radio platform may handle frame retx. 
May help to clarify that `mTimestamp` should be for the last tx attempt?

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

File: include/openthread/platform/radio.h (hunk 1 above)
Comment (bukepo) [reply include/openthread/platform/radio.h:372]: added more description. I think we should require the platform the update the field for each transmit attempt.

File: src/ncp/ncp_base_mtd.cpp [hunk 2]
```diff
@@ -4605,7 +4605,12 @@ void NcpBase::HandlePcapFrame(const otRadioFrame *aFrame, bool aIsTx)
     SuccessOrExit(mEncoder.WriteInt8(-128));    // Noise floor (Currently unused)
     SuccessOrExit(mEncoder.WriteUint16(flags)); // Flags
 
-    SuccessOrExit(mEncoder.OpenStruct()); // PHY-data
+    SuccessOrExit(mEncoder.OpenStruct());                 // PHY-data
+    SuccessOrExit(mEncoder.WriteUint8(aFrame->mChannel)); // Channel
+    SuccessOrExit(
+        mEncoder.WriteUint8(aIsTx ? static_cast<uint8_t>(OT_RADIO_LQI_NONE) : aFrame->mInfo.mRxInfo.mLqi)); // LQI
+    SuccessOrExit(
+        mEncoder.WriteUint64(aIsTx ? aFrame->mInfo.mTxInfo.mTimestamp : aFrame->mInfo.mRxInfo.mTimestamp)); // Timestamp
```
Comment (Irving-cl) [inline src/ncp/ncp_base_mtd.cpp:4613]: Shall we also update the documentation of `spinel.h`.

Channel and LQI seem to be unrelated with this PR... Maybe update the PR description?

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by Vyrastas — APPROVED** [review]
